lib/src/script.dart

```diff
@@ -153,6 +153,21 @@ class Script {
   /// script exits.
   final _outputCloser = StreamCloser<List<int>>();
 
+  /// The script's signal handler to terminate the process.
+  late final Future<FutureOr<bool> Function([ProcessSignal])> _signalHandler;
+
+  /// Sends a [ProcessSignal] to terminate the process.
+  ///
+  /// Defaults to [ProcessSignal.sigterm].
+  ///
+  /// On platforms without signal support, including Windows, the call ignores
+  /// [signal] and the process terminates in a platform specific way.
+  ///
+  /// Returns whether the signal was successfully delivered. Otherwise it
+  /// usually means the process is already dead.
+  FutureOr<bool> signal([ProcessSignal signal = ProcessSignal.sigterm]) async =>
```

Comment:
User-facing methods should never be typed as `FutureOr`, since it's strictly more annoying to deal with than a guaranteed `Future`.

That said, does this really need to be asynchronous at all? We can define "queued up a signal to be delivered once the `fromComponents` callback completes" as "delivered successfully", since it _will be_ delivered and acted on in the future. I'm worried that returning a future will make callers think that this returns once the process has reacted to the signal, which we can't reliably detect.

Making this return synchronously also avoids us needing to think about whether and when this should throw errors.

Reply:
I had made it asynchronous on purpose, but you're right that this may send a misleading guarantee so I made it synchronous now, this unfortunately means that for some scripts the signal handler can't be called ASAP and needs some sort of delay for the futures to resolve